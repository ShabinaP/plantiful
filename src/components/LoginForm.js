import React, { useContext } from "react";
import { Form, FloatingLabel, Button } from "react-bootstrap";
import { ArrowRight } from "phosphor-react";
// import { useNavigate } from "react-router-dom";
import Alert from "./Alert";
import "../styles/LoginForm.css"
import {UserContext} from '../dataContext/UserContext'


const LoginForm = () => {
 const { loginField, handleLogin, handleLoginInputChange,alert, loading, user, isLoggedIn } = useContext(UserContext)
	// let navigate = useNavigate()
	console.log(user)
	console.log(isLoggedIn)
	return (
		<div className="form_wrapper">
				<h2 className="signin_header">Sign in to Plantiful</h2>
				<div className="form_container">
			<Form onSubmit={handleLogin} > 
				<Form.Group className='mb-3' controlId='formBasicUsername'>
						<FloatingLabel
							controlId='floatingInput'
							label='Email address'
							className='mb-3'
						>
							<Form.Control
							    className="w-50"
								type='textbox'
								placeholder="email"
								name='email'
								value={loginField.email}
								onChange={handleLoginInputChange}
								required
							/>
						</FloatingLabel>
					</Form.Group>
				<Form.Group className='mb-3' controlId='formBasicPassword'>
						<FloatingLabel
							controlId='floatingInput'
							label='Password'
							className='mb-3'
						>
							<Form.Control
								type='password'
								className="w-50"
								placeholder="Password"
								data-testid='passwordlogin'
								name='password'
								value={loginField.password}
								onChange={handleLoginInputChange}
								required
							/>
						</FloatingLabel>
					</Form.Group>
					<Button
						type='submit'
						id="submit"
						className='button-primary'
						value={loading ? "Loading..." : "Log in"}
						data-testid='submitbutton'
						style={{ backgroundColor: "#55A356", border: "#013606"}}
					>
						Log in<ArrowRight size={20} />
					</Button>
					<p className="member_text">Not a member?<a href="/registerform" className="signup_link"> Sign up now</a></p>
				<Alert message={alert.message} success={alert.isSuccess} />
			</Form>
			</div>
		</div>
	);
};

export default LoginForm;
