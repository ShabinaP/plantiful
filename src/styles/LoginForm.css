.signin_header {
    display: flex;
    justify-content: center;
    align-items: center;
    color: #013606;
    padding: 15px 10px;
    font-weight: 700;
}

.form_wrapper {
    height: 85vh;
    width: 100%;
    padding: 20px;
    background-color: #F9F5F1;
}

.form_container {
    padding: 20px;
}

.member_text {
    padding: 12px 7px;
}

.signup_link {
    color: #013606;
}