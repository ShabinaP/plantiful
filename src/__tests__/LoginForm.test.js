import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import LoginForm from "../components/LoginForm";

describe("LoginForm component", () => {
	it("renders the basic fields", () => {
		const { asFragment } = render(<LoginForm />);
		expect(asFragment()).toMatchSnapshot();
	});
	it("validates username field", () => {
		render(<LoginForm />);
		const username = screen.getByRole("textbox", { type: "text" });
		userEvent.type(username, "PlantUser");
		expect(username).toHaveValue("PlantUser");
	});
	it("validates password field", () => {
		render(<LoginForm />);
		const password = screen.getByRole("textbox", { type: "password" });
		userEvent.type(password, "P!ant$$$!");
		expect(password).toHaveValue("P!ant$$$!");
	});
	it("submits form", () => {
		const onSubmit = jest.fn();
		const { getByTestId } = render(<LoginForm onSubmit={onSubmit()} />);
		userEvent.click(getByTestId("loginbutton"));
		expect(onSubmit).toHaveBeenCalled();
	});
});
