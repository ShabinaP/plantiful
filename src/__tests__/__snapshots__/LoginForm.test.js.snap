// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders the basic fields 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h2>
        Sign in
      </h2>
    </div>
    <form>
      <div>
        <label
          for="username_input"
        >
          Username
        </label>
        <input
          class="login_form"
          id="loginUsername"
          placeholder="Enter username"
          required=""
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="password_input"
        >
          Password
        </label>
        <input
          class="login_form"
          id="loginPassword"
          placeholder="password"
          required=""
          type="password"
          value=""
        />
      </div>
      <button
        class="button-primary"
        data-testid="loginbutton"
        type="submit"
      >
        Log in
      </button>
    </form>
  </div>
</DocumentFragment>
`;
